testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -10901,6 +10901,26 @@ void checkEndsWith(SqlOperatorFixture f0, FunctionAlias functionAlias) {
             + "requires extra delimiter argument", false);
   }
 
+  /** Tests the {@code SPLIT_PART} operator. */
+  @Test void testSplitPartFunction() {
+    final SqlOperatorFixture f0 = fixture().setFor(SqlLibraryOperators.SPLIT_PART);
+    f0.checkFails("^split_part('hello', 1)^",
+        "No match found for function signature SPLIT_PART\\(<CHARACTER>, <NUMERIC>\\)",
+        false);
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.POSTGRESQL);
+
+    f.checkScalar("SPLIT_PART('abc~@~def~@~ghi', '~@~', 2)", "def", "VARCHAR(15) NOT NULL");
```

Comment:
Do we should return `VARCHAR(15)` for this function? I want to confirm is there a document explaining？

Reply:
This type comes from ARG0_NULLABLE.
It is true that the result will never be larger.
Another reasonable type would be VARCHAR.
I think there are many functions which use this result type inference.
But indeed, the VARCHAR(N) types have some unintuitive behaviors sometimes.

Reply:
Changed to VARCHAR , addressed in latest PR, thanks!